domain/secret/types.go

```diff
@@ -121,3 +122,30 @@ type RemoteSecretInfo struct {
 	CurrentRevision int
 	LatestRevision  int
 }
+
+// SecretMetadata holds metadata about a secret.
+type SecretMetadata struct {
+	// URI is the URI of the secret.
+	URI *secrets.URI
+	// Label is the label of the secret.
+	Label string
+	// Owner is the owner of the secret.
+	Owner secrets.Owner
+}
+
+// SecretRevisionMetadata holds metadata about a secret revision.
+type SecretRevisionMetadata struct {
```

Comment:
This isn't just metadata if it also contains SecretData
Maybe "SecretRevision"

Also, if we're using a bespoke type here, there's no need to embed SecretRevisionMetadata since we don't use all of the fields (eg BackendName). We can just define the 4 attributes we need directly.

Reply:
Accept the BackendName, we use all the rest of the fields, and the finally will be stored in coresecrets.SecretRevisionMetadata. 
It's not worth creating a new struct and then convert to coresecrets.SecretRevisionMetadata later on.

Reply:
Fixed, because we are removing core package dependency from the state.